core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -807,21 +804,22 @@ VariableValue make_variable_value(size_t id, double val) {
   variable_value.value.double_value = val;
   return variable_value;
 }
-Experimental::VariableValue make_variable_value(size_t id, const char* val) {
+Experimental::VariableValue make_variable_value(size_t id,
+                                                const std::string& val) {
   VariableValue variable_value;
-  variable_value.type_id            = id;
-  variable_value.value.string_value = val;
+  variable_value.type_id = id;
+  strncpy(variable_value.value.string_value, val.c_str(),
+          KOKKOS_TOOLS_TUNING_STRING_LENGTH - 1);
   return variable_value;
 }
-SetOrRange make_candidate_set(size_t size, const char** data) {
+SetOrRange make_candidate_set(size_t size, std::string* data) {
   SetOrRange value_set;
-  const char** data_copy = new const char*[size];
+  value_set.set.values.string_value = new TuningString[size];
   for (size_t x = 0; x < size; ++x) {
-    data_copy[x] = new char[strnlen(data[x], 512)]{};
-    strncpy(const_cast<char*>(data_copy[x]), data[x], 512);
+    strncpy((value_set.set.values.string_value[x]), data[x].c_str(),
```

Comment:
why do you have that extra set of parentheses?

Reply:
No good reason, I removed them in b6adc9e